### ISSUE TITLE
Investigation: Loss spikes when loss developes close to convergence

### ISSUE THREAD

**Issue by le1nux** [Modalities/modalities#129]
During the training of 3.6B and 7B with FSDP we experienced a loss spike after the loss as the model was moving towards convergence. 

Things that we should check in our implementation: 

- [x] Correctness of gradient clipping with FSDP
- [x] Exploration of implementational differences of AdamW and Adam in MegetronLM and Pytorch
- [x] Weight initialization
- [ ] ~~GPT2 implementation (we could train a small model directly from Huggingface for comparison)~~

**Comment by le1nux** [comment]
Addressed in PR #143
